From 7d112c91dfa0b0bb0b8280d08f30923e495d2507 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:02 -0400
Subject: [tools] perf/annotate: Add inc_samples method to addr_map_symbol

Message-id: <1408953210-26343-175-git-send-email-jolsa@redhat.com>
Patchwork-id: 88920
O-Subject: [PATCH RHEL7.1 BZ1133083 174/322] perf annotate: Add inc_samples method to addr_map_symbol
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0f4e7a24d0303eaf12847b4fdf4f2d31553fdc61
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Wed Dec 18 16:48:29 2013 -0300

description
===========
Since there are three calls that could receive just the struct
addr_map_symbol pointer and call the symbol method.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index 0615a63..9a20c9e 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -127,8 +127,7 @@ static int perf_report__add_mem_hist_entry(struct perf_tool *tool,
   goto out;
 
  mx = he->mem_info;
- err = symbol__inc_addr_samples(mx->daddr.sym, mx->daddr.map,
-           evsel->idx, mx->daddr.al_addr);
+ err = addr_map_symbol__inc_samples(&mx->daddr, evsel->idx);
  if (err)
   goto out;
 
@@ -190,15 +189,11 @@ static int perf_report__add_branch_hist_entry(struct perf_tool *tool,
      1, 1, 0);
   if (he) {
    bx = he->branch_info;
-   err = symbol__inc_addr_samples(bx->from.sym,
-             bx->from.map, evsel->idx,
-             bx->from.al_addr);
+   err = addr_map_symbol__inc_samples(&bx->from, evsel->idx);
    if (err)
     goto out;
 
-   err = symbol__inc_addr_samples(bx->to.sym,
-             bx->to.map, evsel->idx,
-             bx->to.al_addr);
+   err = addr_map_symbol__inc_samples(&bx->to, evsel->idx);
    if (err)
     goto out;
 
diff --git a/tools/perf/util/annotate.c b/tools/perf/util/annotate.c
index 93614cd..2812e7b 100644
--- a/tools/perf/util/annotate.c
+++ b/tools/perf/util/annotate.c
@@ -503,6 +503,11 @@ int symbol__inc_addr_samples(struct symbol *sym, struct map *map,
  return __symbol__inc_addr_samples(sym, map, notes, evidx, addr);
 }
 
+int addr_map_symbol__inc_samples(struct addr_map_symbol *ams, int evidx)
+{
+ return symbol__inc_addr_samples(ams->sym, ams->map, evidx, ams->al_addr);
+}
+
 static void disasm_line__init_ins(struct disasm_line *dl)
 {
  dl->ins = ins__find(dl->name);
diff --git a/tools/perf/util/annotate.h b/tools/perf/util/annotate.h
index 834b7b5..8de10b1 100644
--- a/tools/perf/util/annotate.h
+++ b/tools/perf/util/annotate.h
@@ -134,6 +134,9 @@ static inline struct annotation *symbol__annotation(struct symbol *sym)
 
 int symbol__inc_addr_samples(struct symbol *sym, struct map *map,
         int evidx, u64 addr);
+
+int addr_map_symbol__inc_samples(struct addr_map_symbol *ams, int evidx);
+
 int symbol__alloc_hist(struct symbol *sym);
 void symbol__annotate_zero_histograms(struct symbol *sym);
 
-- 
1.7.1